client/core/trade.go

```diff
@@ -900,6 +906,34 @@ func (t *trackedTrade) processRedemption(msgID uint64, redemption *msgjson.Redem
 	return nil
 }
 
+// processEpoch processes the new epoch index. If an existing trade has an epoch
+// index < the new index and has a status of OrderStatusEpoch, the order will
+// be assumed to have matched without error, and the status set to executed or
+// booked, depending on the order type.
+//
+// DRAFT NOTE: As of now, if the client is not subscribed to the order book when
+// the order actually matches, the status might not be set correctly. This is a
+// consequence of having no notifications for orders which either are 1)
+// unmatched (market order against empty book, or immediate limit order with
+// restrictive rate), or 2) are booked without matching. This also creates the
+// odd case that a fully-matched standing limit order may be set to
+// OrderStatusBooked before a subsequent match notification might change it
+// to OrderStatusExecuted.
```

Comment:
I'm going to remove this draft note. I don't know if this is an issue or not, but I suspect it is. The crux is that the client has no way of knowing for certain that their order was booked if they don't actually receive a match notification (indicating only partial matching), which will often be the case. The client right now is just assuming that their order was booked, because the epoch expired and they haven't heard otherwise.